### PR TITLE
[test] Modernize QuantumEspresso test

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

**Review by lucamar — APPROVED** [review]
Thanks @teojgo

File: cscs-checks/apps/quantumespresso/quantumespresso_check.py [hunk 1]
```diff
@@ -10,47 +10,57 @@
...
-        self.executable = 'pw.x'
-        self.executable_opts = ['-in', 'ausurf.in',
-                                '-pd', '.true.']
+        self.tags |= {
+            'maintenance' if self.variant == 'maint' else 'production'
+        }
 
-        self.sanity_patterns = sn.all([
+    @sanity_function
+    def assert_simulation_successful(self):
```
Comment (vkarak) [inline cscs-checks/apps/quantumespresso/quantumespresso_check.py:37]:
```suggestion
    def assert_simulation_success(self):
```

**Review by vkarak — COMMENTED** [review]
Would it make sense to parametrize it on the `cpu` vs `gpu` variants as well instead of having to additional classes?

**Comment by teojgo** [comment]
> Would it make sense to parametrize it on the `cpu` vs `gpu` variants as well instead of having to additional classes?

It would need more if blocks, that's why I left it as is, since it already contains some.

**Review by vkarak — APPROVED** [review]

**Pull request merged**